src/coreclr/jit/codegenarm64.cpp

```diff
@@ -5179,10 +5179,17 @@ void CodeGen::genEmitHelperCall(unsigned helper, int argSize, emitAttr retSize,
 
         callTarget = callTargetReg;
 
-        // adrp + add with relocations will be emitted
-        GetEmitter()->emitIns_R_AI(INS_adrp, EA_PTR_DSP_RELOC, callTarget,
-                                   (ssize_t)pAddr DEBUGARG((size_t)compiler->eeFindHelper(helper))
-                                       DEBUGARG(GTF_ICON_METHOD_HDL));
+        if (compiler->opts.compReloc)
+        {
+            // adrp + add with relocations will be emitted
+            GetEmitter()->emitIns_R_AI(INS_adrp, EA_PTR_DSP_RELOC, callTarget,
+                                       (ssize_t)pAddr DEBUGARG((size_t)compiler->eeFindHelper(helper))
+                                           DEBUGARG(GTF_ICON_METHOD_HDL));
+        }
+        else
+        {
+            instGen_Set_Reg_To_Imm(EA_PTRSIZE, callTarget, (ssize_t)addr);
```

Comment:
I think this could be just `instGen_Set_Reg_To_Imm(EA_PTR_DSP_RELOC, ...)` and it will either use addr+add or just emit a 64bit constant inside the function itself. See its impl:

https://github.com/dotnet/runtime/blob/c6e3eb7f6150133ae5491dd401641d7dc25b4486/src/coreclr/jit/codegenarm64.cpp#L2239-L2242

Reply:
Fine to merge as is, just a nit